lib/krb5/test_cc.c

```diff
@@ -691,6 +695,9 @@ main(int argc, char **argv)
 #endif
     test_init_vs_destroy(context, krb5_cc_type_scc);
     test_init_vs_destroy(context, krb5_cc_type_dcc);
+#ifdef HAVE_KEYUTILS_H
+    //test_init_vs_destroy(context, krb5_cc_type_keyring);
```

Comment:
Is this not passing yet?

Reply:
Did I write that? 🤔 I’ll fix now.

Reply:
Actually this test does fail. Investigating.

Reply:
@nicowilliams, this could either be a bug in the test suite, or in the keyring cache. `test_init_vs_destroy()` creates a unique ccache, resolves it into a second, and destroys the first before initializing the second. This leaves the second with a dangling keyring reference to the first.

We could workaround it by catching the error and re-resolving the ccache in `krcc_initialize()`. Or, the tset suite could destroy the first ccache before resolving the second. Thoughts?

Reply:
Interestingly the error I get when trying to access the invalid keyring handle depends on whether I'm running under the debugger or not...

Reply:
This doesn't have to work....

Reply:
I fixed it, if the underlying keyring is destroyed between resolve- and initialize-time, we re-resolve (and thus recreate) it.